src/codemodder/code_directory.py

```diff
@@ -36,13 +36,26 @@ def file_line_patterns(file_path: str | Path, patterns: Sequence[str]):
     ]
 
 
-def filter_files(names: Sequence[str], patterns: Sequence[str], exclude: bool = False):
+def filter_files(
+    names: Sequence[str],
+    parent_path: str,
+    patterns: Sequence[str],
+    exclude: bool = False,
+):
     patterns = (
         [x.split(":")[0] for x in (patterns or [])]
         if not exclude
         # An excluded line should not cause the entire file to be excluded
         else [x for x in (patterns or []) if ":" not in x]
     )
+
+    # TODO: handle case when parent path is "."
+    patterns = [
+        str(Path(parent_path) / Path(pat))
+        if not pat.startswith("*")
```

Comment:
This condition doesn't make sense to me. Do tests fail without it? This means that the behavior is going to be entirely dependent on the presence or absence of a trailing slash in `parent_path`, which I don't think is what we want. If my pattern is `*.py` I want it to mean `/path/to/foo/*.py` and not `/path/to/foo*.py`.

I'm wondering whether removing this also correctly handles the `.` case, although that would also be fixed by normalizing all of the given paths using `os.path.abspath`.